### DISCUSSION TITLE
Gatsby addon for Storybook

### DISCUSSION THREAD

**Discussion by shilman** [gatsbyjs/gatsby#28304] [Ideas / Feature Requests]
Let's make a Gatsby addon for storybook which configures Storybook to work with Gatsby out of the box. On Storybook's side we can modify our CLI to detect that the user is in a gatsby project and install the addon on `sb init`.

Storybook addons can provide `presets`: collections of webpack, babel, and storybook configurations (e.g. for zero-config CRA support). [Thre is presets documentation](https://storybook.js.org/docs/react/api/writing-presets) and the documentation for [how to write an addon](https://www.chromatic.com/blog/how-to-build-a-storybook-addon/) is being rapidly improved by @winkervsbecks. In this case, the addon would consist entirely of a preset and no UI code.

Translating https://www.gatsbyjs.com/docs/visual-testing-with-storybook/ into a preset:

- [ ] `webpackFinal` could be provided by the preset directly
- [ ] The boilderplate `.storybook/preview.js` could be provided by a preview entry
- [ ] Typescript support should be provided out of the box in SB6+
- [ ] `babel-preset-react-app` can be provided by a `babelDefault` entry in the preset, if needed.

cc @wardpeet @KyleAMathews

**Comment by angeloashmore** [comment]
Dropping this comment here with what's working as of today: https://github.com/gatsbyjs/gatsby/issues/26099#issuecomment-783745984

**Comment by angeloashmore** [comment]
[`storybook-addon-gatsby`](https://github.com/prismicio-community/storybook-addon-gatsby) is now available using the latest recommendations in Gatsby's [Visual Testing with Storybook](https://www.gatsbyjs.com/docs/how-to/testing/visual-testing-with-storybook/) document.

**Note**: This addon requires Storybook to be configured for Webpack 5. See the official [Storybook with Webpack 5 guide](https://gist.github.com/shilman/8856ea1786dcd247139b47b270912324) for instructions.

## Install

```bash
npm install -D storybook-addon-gatsby
```

Then include the addon to your Storybook config at `.storybook/main.js`:

```javascript
// .storybook/main.js

module.exports = {
	addons: ["storybook-addon-gatsby"],
};
```
